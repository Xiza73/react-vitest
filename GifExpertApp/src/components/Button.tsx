export interface ButtonProps {
  text: string;
  onClick: (input?: string) => void;
}

export const Button: React.FC<ButtonProps> = ({ text, onClick }) => {
  return <button onClick={() => onClick()}>{text}</button>;
};
