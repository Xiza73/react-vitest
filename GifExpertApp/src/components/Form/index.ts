export * from './Browser';
export * from './Input';
