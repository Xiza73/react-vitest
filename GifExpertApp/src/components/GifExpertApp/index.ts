export * from './CategoriesContainer';
export * from './Gif';
export * from './GifsContainer';
