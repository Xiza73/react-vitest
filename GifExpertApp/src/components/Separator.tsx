export interface SeparatorProps {
  className?: string;
}

export const Separator: React.FC<SeparatorProps> = ({ className }) => (
  <hr className={`w-full my-4 ${className}`} />
);
