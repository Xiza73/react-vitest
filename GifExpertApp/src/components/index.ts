export * from './Form';
export * from './GifExpertApp';
export * from './Button';
