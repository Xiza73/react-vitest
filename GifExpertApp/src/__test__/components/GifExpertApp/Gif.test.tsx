import { Gif } from "@/components";
import { render, screen } from "@testing-library/react";

describe("Tests on <Gif />", () => {
  const url =
    "https://media4.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=cbc864d562njk1yjdf11bs5adtruz31d04dgw2r9md92ug43&rid=giphy.gif&ct=g";
  const title = "One Punch Man";

  test("should show the snapshot", () => {
    const { container } = render(<Gif url={url} title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("should show the image with the correct alt and url", () => {
    render(<Gif url={url} title={title} />);

    const { src, alt } = screen.getByRole<HTMLImageElement>("img");

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
});
