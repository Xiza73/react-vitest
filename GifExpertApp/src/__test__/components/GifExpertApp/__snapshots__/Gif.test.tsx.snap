// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests on <Gif /> > should show the snapshot 1`] = `
<div>
  <article>
    <img
      alt="One Punch Man"
      src="https://media4.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=cbc864d562njk1yjdf11bs5adtruz31d04dgw2r9md92ug43&rid=giphy.gif&ct=g"
    />
  </article>
</div>
`;
